outbound/hmail.js

```diff
@@ -1267,7 +1281,10 @@ class HMailItem extends events.EventEmitter {
             'fails': this.num_failures,
             'rcpts': `${ok_recips.length}/${fail_recips.length}/${bounce_recips.length}`
         });
-        plugins.run_hooks("delivered", this, [host, ip, response, delay, port, mode, ok_recips, secured, authenticated]);
+
+        const hook_params = [host, ip, response, delay, port, mode, ok_recips, secured, authenticated];
+        hook_params.bind_ip = mx.bind;
```

Comment:
You're adding a key to an array. I don't think that'll work.

Reply:
It's Javascript, it works. As noted it's a hack and it's better to change the entire hook args list. Changing the hook arg list means a breaking change I guess. Suggestions are welcome.

Reply:
You can make changes without them being breaking. For example, you can check the arg list length. Better still, pass in objects instead and then test what's being passed in, with an unwrapper function for legacy invocations.